From: Jan Glauber <jglauber@redhat.com>
Subject: [RHEL5 PATCH] s390: inflate spinlock kabi
Date: Tue, 12 Dec 2006 14:25:38 +0100
Bugzilla: 219871
Message-Id: <1165929938.5506.36.camel@bender>
Changelog: s390: inflate spinlock kabi


This feature changes the spinlock type for s390 which would really, really violate
kabi. If we want to add this feature during the RHEL5 lifecycle the only possibility
I see is to inflate the kabi so the feature will not change it.

Jan
-- 
jglauber@redhat.com
jang@de.ibm.com

diff -urNp linux-2.6.18.s390/include/asm-s390/spinlock_types.h linux-2.6.18.s390.diag9c/include/asm-s390/spinlock_types.h
--- linux-2.6.18.s390/include/asm-s390/spinlock_types.h	2006-09-20 05:42:06.000000000 +0200
+++ linux-2.6.18.s390.diag9c/include/asm-s390/spinlock_types.h	2006-12-07 15:06:38.000000000 +0100
@@ -7,15 +7,16 @@
 
 typedef struct {
 	volatile unsigned int lock;
+	/* dummy field to prepare kabi for diag9c */
+	volatile unsigned int owner_pc;
 } __attribute__ ((aligned (4))) raw_spinlock_t;
 
-#define __RAW_SPIN_LOCK_UNLOCKED	{ 0 }
+#define __RAW_SPIN_LOCK_UNLOCKED	{ 0, 0 }
 
 typedef struct {
 	volatile unsigned int lock;
-	volatile unsigned int owner_pc;
 } raw_rwlock_t;
 
-#define __RAW_RW_LOCK_UNLOCKED		{ 0, 0 }
+#define __RAW_RW_LOCK_UNLOCKED		{ 0 }
 
 #endif

